L83-core-cpp-tls-key-export-feature.md

```diff
@@ -0,0 +1,93 @@
+Title
+----
+* Author(s): Vignesh Babu (Vignesh2208)
+* Approver: ctiller
+* Status: In Review
+* Implemented in: core, cpp
+* Last updated: 07/29/2021
+* Discussion at: https://github.com/grpc/grpc/issues/24944
+
+## Abstract
+
+This proposal discusses additions to the core/cpp public headers to support exporing of TLS Keys on the client and server side to aid in decryption of packet captures using tools such as Wireshark.
+
+## Background
+
+Currently unlike the GRPC Go implementation, GRPC C++ does not allow export of TLS keys at run time. This is needed to decrypt and analyse packet captures for debugging purposes. The keys when exported in files in a specific format, can be read by debugging tools like wireshark and they can automatically decrypt packet captures.
+
+### Related Proposals:
+
+N/A
+
+## Proposal
+
+The proposal extends the existing `grpc::experimental::TlsCredentialsOptions` class with methods to set configuration for supporting Tls Key export at run-time. The proposed changes
+include:
+
+* A `grpc_tls_key_log_format` enum is added to `include/grpc/grpc_security_constants.h`
```

Comment:
Let's omit this for now.

Instead, add a channel arg (string) for the path, and if we need to also specify the format we can add a secondary parameter in the future.

Reply:
Sure. Should we get rid of the struct grpc::experimental::TlsKeyLoggerConfig as well ? Currently it has two members: the file path string and file log format. Getting rid of the file log format would leave it with just one member. I could get rid of the struct but it would make it harder to add any new extension in the future.

Reply:
One benefit I can think of using channel args is that it is agnostic to credential types and thus can support both SSL and TLS credentials. However, I do not think that benefit outweighs all three reasons you mentioned here, especially there is no motivation to support the feature in the old SSL stack.

Reply:
I think we definitely want this feature to be configured via TlsCreds, not via channel args.  That's how we're doing all credential configuration, and I think it would be quite strange to do this one feature differently.